src/Components/Components/src/Reflection/ComponentProperties.cs

```diff
@@ -41,7 +42,7 @@ public static void SetProperties(in ParameterView parameters, object target)
                 foreach (var parameter in parameters)
                 {
                     var parameterName = parameter.Name;
-                    if (!writers.WritersByName.TryGetValue(parameterName, out var writer))
+                    if (!writers.TryGetValue(parameterName, out var writer))
```

Comment:
One additional optimization we could make is to stash the the `WritesForType` as an internal property on `ComponentBase` and avoid the dictionary lookup there. 

```C#
if (target is ComponentBase component)
{
  component.Writer ?? = new WritersForType(targetType);
  _writer = component.Writer;
}
else
{
 // old code
}
```

Reply:
Thanks for the suggestion. Since this didn't show up in the profile traces as a hot path, I'd prefer not to add an optimization for it.

I guess that doing the lookup for this dictionary is relatively cheap because `component.GetHashCode()` is much cheaper than computing a string hash.